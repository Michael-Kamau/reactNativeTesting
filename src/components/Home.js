import React,{Component} from 'react';
import {View,Text,TextInput}  from 'react-native';
import Profile from './Profile';
class Home extends Component{

    constructor(){
        super();
        this.state={
            data:"test",
        }
    }




    change(x){
        this.setState({data:x*5})
    }


    render(){
        return(
            <View>
                <Text>Home Component</Text>
                <Profile/>
            </View>
        )
    }
}



export default Home;


