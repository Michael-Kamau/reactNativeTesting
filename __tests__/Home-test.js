import 'react-native';
import React from 'react';
import Home from '../src/components/Home'
import renderer from 'react-test-renderer'

test('Home snapshot',()=>{
    const snap= renderer.create(
        <Home />
    ).toJSON();


    expect(snap).toMatchSnapshot();
});
