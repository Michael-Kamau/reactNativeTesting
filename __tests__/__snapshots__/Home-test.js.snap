// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home snapshot 1`] = `
<View>
  <Text>
    Home Component
  </Text>
  <View>
    <Text>
      Profile Component
    </Text>
  </View>
</View>
`;
